xarray/core/rolling_exp.py

```diff
@@ -32,6 +33,20 @@ def move_exp_nanmean(array, *, axis, alpha):
         return numbagg.move_exp_nanmean(array, axis=axis, alpha=alpha)
 
 
+def move_exp_nansum(array, *, axis, alpha):
+    if is_duck_dask_array(array):
+        raise TypeError("rolling_exp is not currently support for dask-like arrays")
+    import numbagg
+
+    if LooseVersion(numbagg.__version__) < LooseVersion("0.2.0"):
+        raise ValueError("`rolling_exp(...).sum() requires numbagg>=0.2.0.")
+
+    if axis == ():
```

Comment:
I think we need this because `ds.reduce` calls the reduction function with `axis=()` on variables that aren't over the supplied dim.

I don't remember why that is. To the extent we can skip over those variables rather than manage that in these lower functions, that would be simpler. Lmk if anyone is familiar with this!

Reply:
I think this indeed skips variables that don't have the supplied dim. But it can change the dtype so - should it only return `array`?

`sum` uses `self.obj.reduce` which passes all `data_vars` - I am not sure the alternative would be easier.

Reply:
Yes, I can't remember why we change the type (I copied this from `move_exp_nanmean` which I did a couple of years ago) — possibly because we want a consistent return type, which has some merit.

I'll look more at the axis arg. It does seem out of place here — maybe either it should be in numbagg or handled higher in the call chain

Reply:
This is fixed now, thanks @mathause for raising